yaserde_derive/src/ser/expand_struct.rs

```diff
@@ -176,6 +176,54 @@ pub fn serialize(
               })
             }
           }
+          FieldType::FieldTypeStruct { .. } => {
+            if let Some(ref d) = field_attrs.default {
+              let default_function = Ident::new(&d, field.span());
+              Some(quote! {
+                let struct_start_event = if let Some(ref value) = self.#label {
+                  if *value != #default_function() {
+                    struct_start_event.attr(#label_name, &*{
+                      use std::mem;
+                      match yaserde::ser::to_string_content(value) {
+                        Ok(value) => {
+                          unsafe {
+                            let ret : &'static str = mem::transmute(&value as &str);
```

Comment:
I'm agreed it's better for performances, but it can be simplest to wrote something like:
```rust
struct_start_event.attr(#label_name, &{
  match yaserde::ser::to_string_content(value) {
    Ok(value) => {
      value.clone()
    }
    Err(msg) => return Err("Unable to serialize content".to_owned()),
  }
}
```

(I have not tested ...)
For me the difference is:
- in your proposal, we have a forget memory, which was not really good.
- in my remark, we copy the object...

Reply:
Actually I've just copied it from another case. Now there are 8 such places with unsafe code that can be refactored like you say. This is probably out of scope of current PR. I have a plan to make a small refactoring PR and can include this changes into it.

Reply:
Hm.. The problem here is that we are trying to borrow a temporary that is being dropped. It can be easily fixed, but need to make it in all 8 places.

Reply:
Okay so I'll accept the current PR, and I create an issue to improve that.
I'll assign to you if it's fine.